solidity.pegjs

```diff
@@ -1424,8 +1456,18 @@ SourceElements
     }
 
 SourceElement
-  = AssignmentExpression __ EOS
-  / DeclarativeExpression __ EOS
+  = exp:AssignmentExpression  __ EOS &{
+      return (
+        exp.type === 'AssignmentExpression' && 
+        exp.left.type === 'DeclarativeExpression' && 
+        exp.left.storage_location === null
+      )
+    }{

```

Comment:
So get rid of `AssignmentExpression` up there and replace with a `StateVariableDeclaration` that has an optional `= expression`, visibility and constant specifiers, no storage_location field, a `value` field that may be an expression or null, and don't preserve the equality operator?